### DISCUSSION TITLE
Show submitted strategies' keys in authenticate method

### DISCUSSION THREAD

**Discussion by ikadgzl** [sergiodxa/remix-auth#225] [Ideas]
So, whenever I submitted strategy using use method in Authenticator, such as:

```
authenticator.use(
  new FormStrategy(async ({ form }) => {
  }),
  "user-pass"
);
```

I'd like authenticate method in Authenticator to show keys of whatever strategies I've submitted in the first parameter which is _strategy:string_.

So basically, I set two strategy, one is called `"user-pass"` the other is `"otp"`

Then in somewhere, I call `authenticator.authenticate(""),` now at this point, I'd like typescript to guide me on what to write for the first parameter as if I would define a string literal type, like:

>  type StrageyNames = 'user-pass' | 'otp'

but generated dynamically based on strategies set before, is it possible?

**Comment by sergiodxa** [comment]
Because you can add and remove strategies at runtime, there's no way AFAIK to have them typed in TS.
